tock/hours/models.py

```diff
@@ -108,6 +110,21 @@ def notes_list(self):
         return self.notes.split('\n')
 
     def save(self, *args, **kwargs):
+        """Custom save() method to append employee grade info to each
+        TimecardObject. Grades are only appended if they are in force prior to
+        the end of the reporting period for which the timecard object is being
+        filed. Also updates timecard object with the submitted status of the
+        related timecard."""
+        emp_grd_objects = EmployeeGrade.objects.filter(
+                Q(employee = self.timecard.user)
+                & Q(g_start_date__lte = self.timecard.reporting_period.end_date)
+                ).all().aggregate(Max('g_start_date'))['g_start_date__max']
+
+        if emp_grd_objects:
+            self.grade = EmployeeGrade.objects.filter(
+                employee=self.timecard.user,
+                g_start_date=emp_grd_objects)[0]
+
```

Comment:
The code to figure out which grade applies to a given employee should live in a model method on UserData--I'd recommend setting it to default to their current grade, with optional arguments to give their grade as of a specific date. Then call that method from here to figure out what grade applies to the relevant time card.

The logic here is that it makes that code more accessible if we need it in the future, and it makes it easy to access a fact associated with the employee (their grade at a given time) from the employee's userdata object, where we're used to finding that info.